File: week1/problems/mario_less.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    // Ask user for height (1 to 8 inclusive)
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    int hash = 1;
    int space = height - 1;

    for (int i = 0; i < height; i++)
    {
        // Print spaces
        for (int j = 0; j < space; j++)
        {
            printf(" ");
        }

        // Print hashes
        for (int k = 0; k < hash; k++)
        {
            printf("#");
        }
        printf("\n");

        // Sequence for hash and space
        hash++;
        space--;
    }
}

File: week6/problems/mario-less.py
from cs50 import get_int


def main():
    height = get_height()

    # Print pyramid
    for i in range(1, height + 1):
        build(" ", height - i)
        build("#", i)
        print()


# Get height from user between 1 and 8 inclusive
def get_height():
    while True:
        n = get_int("Height: ")
        if n > 0 and n < 9:
            break
    return n


# Print characters n times in one line
def build(char, n):
    for i in range(n):
        print(char, end="")


if __name__ == "__main__":
    main()